File: Solutions/3200-3299/3217. Delete Nodes From Linked List Present in Array/Solution.java
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        HashMap<Integer,Boolean> map = new HashMap<>();
        for(int num: nums)
        {
            map.put(num,true);
        }
        
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        
        ListNode prev=dummy;
        ListNode cur=head;
        
        while(cur!=null)
        {
            if(map.containsKey(cur.val))
            {
                prev.next=cur.next;
                cur=prev.next;
            }
            else
            {
                prev=cur;
                cur=cur.next;
            }
        }
        
        return dummy.next;
    }
}

File: Solutions/3200-3299/3217. Delete Nodes From Linked List Present in Array/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        nums = set(nums)
        res = ListNode(0, head)

        prev = res
        while prev.next:
            if prev.next.val in nums:
                prev.next = prev.next.next
            else:
                prev = prev.next
        
        return res.next